__tests__/PetClass.test.js

```diff
@@ -0,0 +1,226 @@
+  HUNGER_DECREMENT
+} = petConfig;
+
+describe('Pet class', () => {
+  it('returns an object', () => {
+    expect(new Pet()).toBeInstanceOf(Object);
+  });
+
+  it('returns an instance of Pet', () => {
+    expect(new Pet()).toBeInstanceOf(Pet);
+  });
+
+  it('throws TypeError if the name argument is not <string>', () => {
+    expect(() => new Pet(42)).toThrow(TypeError);
+    expect(() => new Pet(false)).toThrow('name must be a string or undefined');
+  });
+});
+
+describe('Pet instance', () => {
+  it('name initialised as name argument (if provided)', () => {
+    expect(new Pet('Ollie').name).toBe('Ollie');
+  });
+
+  it('name initialised as undefined (if name argument not provided)', () => {
+    expect(new Pet().name).toBeUndefined();
+  });
+
+  it('age initialised as AGE_INIT', () => {
+    expect(new Pet().age).toBe(AGE_INIT);
+  });
+
+  it('hunger intitialised as HUNGER_INIT', () => {
+    expect(new Pet().hunger).toBe(HUNGER_INIT);
+  });
+
+  it('fitness initialised as FITNESS_INIT', () => {
+    expect(new Pet().fitness).toBe(FITNESS_INIT);
+  });
+});
+
+describe('Pet prototype', () => {
+  it('has growUp method', () => {
+    expect(Pet.prototype.growUp).toBeInstanceOf(Function);
+  });
+
+  it('has walk method', () => {
+    expect(Pet.prototype.walk).toBeInstanceOf(Function);
+  });
+
+  it('has feed method', () => {
+    expect(Pet.prototype.feed).toBeInstanceOf(Function);
+  });
+
+  it('has checkUp method', () => {
+    expect(Pet.prototype.checkUp).toBeInstanceOf(Function);
+  });
+
+  it('has isAlive getter method', () => {
+    expect(
+      Object.getOwnPropertyDescriptor(Pet.prototype, 'isAlive').get
+    ).toBeInstanceOf(Function);
+  });
+});
+
+describe('growUp method', () => {
+  it('throws exception when isAlive returns false', () => {
+    const pet = new Pet();
+    pet.age = AGE_MAX;
+    expect(() => pet.growUp()).toThrow('Your pet is no longer alive');
+  });
+
+  it('increments age by AGE_INCREMENT', () => {
+    const pet = new Pet();
+    pet.growUp();
+    expect(pet.age).toBe(AGE_INCREMENT);
+
+    pet.growUp();
+    expect(pet.age).toBe(AGE_INCREMENT * 2);
+  });
+
+  it('increments hunger by HUNGER_INCREMENT', () => {
+    const pet = new Pet();
+    pet.growUp();
+    expect(pet.hunger).toBe(HUNGER_INIT + HUNGER_INCREMENT);
+
+    pet.growUp();
+    expect(pet.hunger).toBe(HUNGER_INIT + HUNGER_INCREMENT * 2);
+  });
+
+  it('decrements fitness by FITNESS_DECREMENT', () => {
+    const pet = new Pet();
+    pet.growUp();
+    expect(pet.fitness).toBe(FITNESS_INIT - FITNESS_DECREMENT);
+
+    pet.growUp();
+    expect(pet.fitness).toBe(FITNESS_INIT - 2 * FITNESS_DECREMENT);
+  });
+});
+
+describe('walk method', () => {
+  it('throws exception when isAlive returns false', () => {
+    const pet = new Pet();
+    pet.age = AGE_MAX;
+    expect(() => pet.walk()).toThrow('Your pet is no longer alive');
+  });
+
+  it('increments fitness by FITNESS_INCREMENT, but is clamped to FITNESS_INIT', () => {
```

Comment:
Would it not be more appropriate to break this up into 2 tests? You are actually testing 2 behaviours here, firstly that the fitness is increased by the correct increment, and secondly that it does not go beyond the allowed range. Its a minor difference but its always good to make unit tests as granular as possible

It would be good to add extra tests in this way for the other methods here too as the same approach is being taken there too